tests/system/feature_store/test_spark_engine.py

```diff
@@ -713,6 +721,241 @@ def test_schedule_on_filtered_by_time(self, partitioned):
         assert len(resp.to_dataframe() == 4)
         assert "uri" not in resp.to_dataframe() and "katya" not in resp.to_dataframe()
 
+    def test_stream_aggregation(self):
+        name = f"measurements_{uuid.uuid4()}"
+        test_base_time = datetime.fromisoformat("2020-07-21T21:40:00+00:00")
+
+        schema = StructType(
+            [
+                StructField("time", TimestampType(), True),
+                StructField("first_name", StringType(), True),
+                StructField("last_name", StringType(), True),
+                StructField("bid", IntegerType(), True),
+                StructField("mood", StringType(), True),
+            ]
+        )
+
+        data = (
+            {
+                "time": [
+                    test_base_time,
+                    test_base_time + pd.Timedelta(minutes=7),
+                    test_base_time + pd.Timedelta(minutes=14),
+                    test_base_time + pd.Timedelta(minutes=21),
+                    test_base_time + pd.Timedelta(minutes=28),
+                ],
+                "first_name": ["moshe", "yosi", "yosi", "moshe", "yosi"],
+                "last_name": ["cohen", "levi", "levi", "cohen", "levi"],
+                "bid": [2000, 10, 11, 12, 16],
+                "mood": ["bad", "good", "bad", "good", "good"],
+            },
+            {
+                "time": [
+                    test_base_time + pd.Timedelta(minutes=35),
+                    test_base_time + pd.Timedelta(minutes=42),
+                ],
+                "first_name": ["moshe", "yosi"],
+                "last_name": ["cohen", "levi"],
+                "bid": [20, 40],
+                "mood": ["bad", "good"],
+            },
+        )
+
+        checkpoint_path = "/tmp/checkpoints/"
+        path = f"{self.output_dir(url=False)}/test_aggregations_parquet/"
+        fsys = fsspec.filesystem(
+            "file" if self.run_local else v3iofs.fs.V3ioFS.protocol
+        )
+        try:
+            fsys.rm(checkpoint_path, recursive=True)
+        except FileNotFoundError:
+            pass
+
+        name_spark = f"{name}_spark"
+
+        for i in range(2):
+            # Cleanup parquet directory
+            try:
+                fsys.rm(path, recursive=True)
+            except FileNotFoundError:
+                pass
+            fsys.makedirs(path)
+
+            # Create Spark DataFrame
+            # Convert the dictionary to a list of rows
+            rows = list(
+                zip(
+                    data[i]["time"],
+                    data[i]["first_name"],
+                    data[i]["last_name"],
+                    data[i]["bid"],
+                    data[i]["mood"],
+                )
+            )
+            spark = SparkSession.builder.appName("test").getOrCreate()
```

Comment:
Only remote spark will work in CI. Local spark will not work because Java isn't even installed on the image that runs the tests.